eng/code-quality-reports/src/main/java/com/azure/tools/checkstyle/checks/ThrowFromClientLoggerCheck.java

```diff
@@ -82,8 +58,14 @@ public void visitToken(DetailAST token) {
                 if (!LOGGER_LOG_AND_THROW.equals(methodCall)) {
                     return;
                 }
+
+                if (!isValidScope(token)) {
+                    return;
+                }
+
                 DetailAST exprToken = token.getParent();
-                if (exprToken.getNextSibling().getType() != TokenTypes.LITERAL_RETURN) {
+                if (exprToken.getNextSibling() == null || exprToken.getNextSibling().getNextSibling() == null
```

Comment:
What's the reason for this logic? Might need some clarifying docs.
(ie. Why don't we have `exprToken.getNextSibling().getNextSibling().getNextSibling()` if we have `exprToken.getNextSibling().getNextSibling()`?)

Would: TokenUtils.findFirstTokenByPredicate(DetailAST root, Predicate<DetailAST> predicate) help?

Reply:
The logic here is checking immediately return after throw statements. Any of getNextSibling return null implies no return statement immediately after throw

Reply:
What about `exprToken.getNextSibling().getNextSibling() == null`?

Reply:
Document this in the code.

Reply:
exprToken.getNextSibling() could be SEME, ';'.
exprToken.getNextSibling().getNextSibling() could be null if there is no other statement following ; https://github.com/Azure/azure-sdk-for-java/blob/0c12ca0d946f943531503451549610bad658ca82/sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventDataBatch.java#L198